File: backend/voting-app/src/main/java/com/tu/votingapp/dto/response/elections/ElectionResponseDTO.java
package com.tu.votingapp.dto.response.elections;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ElectionResponseDTO {
    private Long id;
    private String electionName;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;
    private String electionType; // enum.name()
    private String status;       // enum.name()
    private Long createdById;
    private List<CandidateResponseDTO> candidates;
    private List<PartyResponseDTO> parties;
}

File: frontend/android/app/src/main/java/com/example/android/dto/response/elections/ElectionResponseDTO.kt
package com.example.android.dto.response.elections

import kotlinx.serialization.Serializable

@Serializable
data class ElectionResponseDTO(
    val id: Long,
    val electionName: String,
    val description: String?,
    val startDate: String,
    val endDate: String,
    val electionType: String,
    val status: String,
    val createdById: Long,
    val candidates: List<CandidateResponseDTO>,
    val parties: List<PartyResponseDTO>
)